GCRCatalogs/dc2_dia_object.py

```diff
@@ -0,0 +1,107 @@
+"""
+DC2 DIA Object Catalog Reader
+"""
+
+import os
+import re
+import warnings
+
+import numpy as np
+
+from .dc2_dm_catalog import (DC2DMCatalog,
+                             convert_flux_to_nanoJansky,
+                             convert_nanoJansky_to_mag,
+                             convert_flux_err_to_mag_err,
+                             create_basic_flag_mask)
+
+__all__ = ['DC2DiaObjectCatalog']
+
+
+class DC2DiaObjectCatalog(DC2DMCatalog):
+    r"""DC2 DIA Object Catalog reader
+
+    Parameters
+    ----------
+    base_dir          (str): Directory of data files being served, required
+    filename_pattern  (str): The optional regex pattern of served data files
+    use_cache        (bool): Cache read data in memory
+    is_dpdd          (bool): File are already in DPDD-format.  No translation.
+
+    Attributes
+    ----------
+    base_dir          (str): The directory of data files being served
+
+    Notes
+    -----
+    """
+    # pylint: disable=too-many-instance-attributes
+    FILE_DIR = os.path.dirname(os.path.abspath(__file__))
+    FILE_PATTERN = r'dia_object_\d+\.parquet$'
+    SCHEMA_FILENAME = 'schema.yaml'
+    META_PATH = os.path.join(FILE_DIR, 'catalog_configs/_dc2_dia_object_meta.yaml')
+
+    @staticmethod
+    def _generate_modifiers(dm_schema_version=3, bands='ugrizy'):
```

Comment:
the `bands` argument does not appear in the base class. Is the intention to overwrite the call signature? If so, add `# pylint: disable=arguments-differ` and also update the docstring.

Reply:
Yes, we will need `bands` for [DIA]Object tables.

Reply:
However, we don't need `dm_schema_version` for DIAObject.  It's not used.  Should I:
A) Leave that in?
B) Take it out, leaving just `bands='ugrizy'`, which leaves the risk of a positional argument call getting confused and overriding the default `bands` with, e.g., `3` (for the dm_schema_version)?
C) Use `def _generate_modifiers(*, bands='ugrizy')` to ignore any positional arguments?

Reply:
Is the reason that we don't need `dm_schema_version` because now we hard code the new flux name in this reader? That is ok, and I think Approach A (which you implemented) is fine (and maybe in the future there is other name changes...). But maybe it should raise a warning or error when people try to specify `dm_schema_version < 3`?

Reply:
`instFlux` will very like remain the internal pipelines name, and there is no DIA DPDD product that we have or will generate for DM Schema Version <= 3 (I would argue that they're really actually all >=4, so I'll change that default to 4).  It is possible that it could change so keep it makes sense.

So (A) and a warning makes sense.  Thanks for the suggestion for a warning.